src/votekit/elections/election_types/ranking/stv.py

```diff
@@ -0,0 +1,406 @@
+        return False
+
+    def _simultaneous_elect_step(
+        self, profile: PreferenceProfile, prev_state: ElectionState
+    ) -> tuple[tuple[frozenset[str], ...], PreferenceProfile]:
+        """
+        Run one step of an election from the given profile and previous state.
+        Used for simultaneous STV election if candidates cross threshold.
+
+        Args:
+            profile (PreferenceProfile): Profile of ballots.
+            prev_state (ElectionState): The previous ElectionState.
+
+        Returns:
+            tuple[tuple[frozenset[str],...], PreferenceProfile]:
+                A tuple whose first entry is the elected candidates, ranked by first-place votes,
+                and whose second entry is the profile of ballots after transfers.
+        """
+        ranking_by_fpv = prev_state.remaining
+
+        # find ranking of cands above threshold
+        elected = []
+        for s in ranking_by_fpv:
+            c = list(s)[0]  # all cands in set have same score
+            if prev_state.scores[c] >= self.threshold:
+                elected.append(s)
+
+            # since ranking is ordered by fpv, once below threshold we are done
+            else:
+                break
+
+        # no tiebreaks needed here, since threshold is such that
+        # no more than m candidates can be elected
+
+        ballots_by_fpv = ballots_by_first_cand(profile)
+        new_ballots = [Ballot()] * profile.num_ballots
+        ballot_index = 0
+
+        # for candidate sets above threshold
+        for s in elected:
+            for candidate in s:
+                # transfer ballots with candidate in first, using transfer method
+                transfer_ballots = self.transfer(
+                    candidate,
+                    prev_state.scores[candidate],
+                    ballots_by_fpv[candidate],
+                    self.threshold,
+                )
+                new_ballots[
+                    ballot_index : (ballot_index + len(transfer_ballots))
+                ] = transfer_ballots
+                ballot_index += len(transfer_ballots)
+
+        # preserve remaining ballots
+        for candidate in set([c for s in ranking_by_fpv for c in s]).difference(
+            [c for s in elected for c in s]
+        ):
+            transfer_ballots = tuple(ballots_by_fpv[candidate])
+            new_ballots[
+                ballot_index : (ballot_index + len(transfer_ballots))
+            ] = transfer_ballots
+            ballot_index += len(transfer_ballots)
+
+        # remove elected from all ballots
+        cleaned_ballots = remove_cand(
+            [c for s in elected for c in s],
+            tuple([b for b in new_ballots if b.ranking]),
+        )
+
+        remaining_cands = set(profile.candidates).difference(
+            [c for s in elected for c in s]
+        )
+        new_profile = PreferenceProfile(
+            ballots=cleaned_ballots, candidates=tuple(remaining_cands)
+        )
```

Comment:
I would remove most of these comments and try to describe the process in the doc string. The no tiebreak comment is good to keep